plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/RequestSampler.java

```diff
@@ -44,15 +51,20 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks
     ) {
-
         final String trace = attributes.get(AttributeKey.stringKey(TRACE));
 
         if (trace != null) {
             return (Boolean.parseBoolean(trace) == true) ? SamplingResult.recordAndSample() : SamplingResult.drop();
-        } else {
-            return defaultSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
         }
 
+        for (String samplerName : this.samplerList) {
```

Comment:
I think we should'd do chaining on the SamplingDecision.RECORD_ONLY as it has it's own distinct definition - 

**SamplingDecision.RECORD_ONLY**

Span is recorded only. The resulting span will record all information like timings and attributes but will not be exported. Downstream parent-based samplers will not sample the span.